Lib/posixpath.py

```diff
@@ -135,6 +135,35 @@ def splitdrive(p):
     return p[:0], p
 
 
+def splitroot(p):
+    """Split a pathname into drive, root and tail. On Posix, drive is always
+    empty; the root may be empty, a single slash, or two slashes. The tail
+    contains anything after the root. For example:
+
+        splitroot('foo/bar') == ('', '', 'foo/bar')
+        splitroot('/foo/bar') == ('', '/', 'foo/bar')
+        splitroot('//foo/bar') == ('', '//', 'foo/bar')
+        splitroot('///foo/bar') == ('', '/', '//foo/bar')
+    """
+    p = os.fspath(p)
+    if isinstance(p, bytes):
+        sep = b'/'
+        empty = b''
+    else:
+        sep = '/'
+        empty = ''
+    if p[:1] != sep:
+        # Relative path, e.g.: 'foo'
+        return empty, empty, p
+    elif p[1:2] != sep or p[2:3] == sep:
+        # Absolute path, e.g.: '/foo', '///foo', '////foo', etc.
+        return empty, sep, p[1:]
```

Comment:
Could similarly use `str.partition()` here as well:

```suggestion
        return p.partition(sep)
```

Reply:
I think that could obscure what the code is doing slightly. Firstly because the code above and below doesn't use `partition()`, and secondly because the astute reader might ask "why use partition() if you already know the position of the separator?" and conclude they're missing something.

Reply:
Yes, this one isn't nearly as much of a "simplification" as the other. Better to leave it as you have it currently, as you say :))